internal/repositories/github/service.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2024 Stacklok, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package github contains logic relating to the management of github repos
+package github
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/uuid"
+	"github.com/rs/zerolog/log"
+
+	"github.com/stacklok/minder/internal/db"
+	"github.com/stacklok/minder/internal/events"
+	"github.com/stacklok/minder/internal/logger"
+	ghprovider "github.com/stacklok/minder/internal/providers/github"
+	ghclient "github.com/stacklok/minder/internal/repositories/github/clients"
+	"github.com/stacklok/minder/internal/repositories/github/webhooks"
+)
+
+// RepositoryService encapsulates logic related to registering and deleting repos
+type RepositoryService interface {
+	// DeleteRepositoryByName removes the webhook and deletes the repo from the
+	// database. The repo is identified by its name and project.
+	DeleteRepositoryByName(
+		ctx context.Context,
+		client ghclient.GitHubRepoClient,
+		projectID uuid.UUID,
+		repoOwner string,
+		repoName string,
```

Comment:
I like how this interface is only in terms of other interfaces or dumb data objects.  Nice!